yb-voyager/src/tgtdb/oracle.go

```diff
@@ -645,34 +645,65 @@ func (tdb *TargetOracleDB) MaxBatchSizeInBytes() int64 {
 	return utils.GetEnvAsInt64("MAX_BATCH_SIZE_BYTES", int64(2)*1024*1024*1024) //default: 2 * 1024 * 1024 * 1024 2GB
 }
 
-func (tdb *TargetOracleDB) GetIdentityColumnNamesForTable(tableNameTup sqlname.NameTuple, identityType string) ([]string, error) {
-	sname, tname := tableNameTup.ForCatalogQuery()
-	query := fmt.Sprintf(`Select COLUMN_NAME from ALL_TAB_IDENTITY_COLS where OWNER = '%s'
-	AND TABLE_NAME = '%s' AND GENERATION_TYPE='%s'`, sname, tname, identityType)
-	log.Infof("query of identity(%s) columns for table(%s): %s", identityType, tableNameTup, query)
-	var identityColumns []string
-	err := tdb.WithConnFromPool(func(conn *sql.Conn) (bool, error) {
-		rows, err := conn.QueryContext(context.Background(), query)
+func (tdb *TargetOracleDB) GetIdentityColumnNamesForTables(tableNameTuples []sqlname.NameTuple, identityType string) (*utils.StructMap[sqlname.NameTuple, []string], error) {
+	result := utils.NewStructMap[sqlname.NameTuple, []string]()
+	if len(tableNameTuples) == 0 {
+		return result, nil
+	}
+
+	// Build table name list and lookup map
+	var tableNames []string
+	tableNameMap := make(map[string]sqlname.NameTuple) // map to lookup table name tuples by table name
+
+	for _, t := range tableNameTuples {
+		_, tableName := t.ForCatalogQuery()
+		tableNames = append(tableNames, fmt.Sprintf("'%s'", tableName))
+		tableNameMap[tableName] = t
+	}
+
+	query := fmt.Sprintf(`
+		SELECT TABLE_NAME, COLUMN_NAME
+		FROM ALL_TAB_IDENTITY_COLS
+		WHERE TABLE_NAME IN (%s)
+		  AND GENERATION_TYPE = '%s'
+		ORDER BY TABLE_NAME, COLUMN_NAME`,
```

Comment:
OWNER filtration `Where OWNER = '<schema>'`  is required to only fetch the tables from that owner in case there are same tables in multiple schemas

Reply:
Good catch. Bad miss.